dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumContextProvider.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.webview
+
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.domain.RumContext
+import com.datadog.android.rum.internal.domain.scope.RumApplicationScope
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.monitor.DatadogRumMonitor
+
+internal class RumContextProvider {
```

Comment:
Nice trick, the only downside is that the check is performed outside of this class meaning that multiple calls to getRumContext could still happen and you'd log the issue every time. 

Instead you could have a boolean in here to remember if you already sent an internal log.

Reply:
yeah but this will mix a bit the logic, if you think about it there will not be too many classes as there (one/webview) only ...and this will mean that say you have 3 WebViews in the same time : you will have max 3 logs.